### PR TITLE
Adjust for maven archetype 3.x

### PR DESCRIPTION
**Pull request by laeubi**

### CODE REVIEW

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java [hunk 1]
```diff
@@ -178,17 +179,25 @@ public void scheduled(IJobChangeEvent event) {
...
-    archetype.setVersion("1.0");
-    archetype.setRepository(new File("repositories/testrepo").getAbsoluteFile().toURI().toASCIIString());
-    Collection<IProject> projects = createProjectsFromArchetype("mngeclipse-2110", new MavenArchetype(archetype), null);
-    assertEquals(1, projects.size());
-    IProject project = projects.iterator().next();
-    assertTrue(project.isAccessible());
+    File templateSettings = new File("settings_archetype_repo.xml").getCanonicalFile().getAbsoluteFile();
+    String interpolated = Files.readString(templateSettings.toPath()).replace("${repoUrl}",
+        new File("repositories/testrepo").getAbsoluteFile().toURI().toASCIIString());
+    File tempFile = File.createTempFile("settings", ".xml");
```
Comment (HannesWell) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java:185]: If you'd use [Junit-4's `TemporaryFolder` rule](https://junit.org/junit4/javadoc/latest/org/junit/rules/TemporaryFolder.html) the try-finally block would not be necessary and we have less set-up/tear-down within the test-case method itself.

**Review by HannesWell — COMMENTED** [review]
Thanks for your effort in making this test work. I think it is good to have it.

It was necessary because of [ARCHETYPE-519](https://issues.apache.org/jira/browse/ARCHETYPE-519), wasn't it?

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java [hunk 2]
```diff
@@ -178,17 +179,25 @@ public void scheduled(IJobChangeEvent event) {
...
-    archetype.setGroupId("org.eclipse.m2e.its");
-    archetype.setArtifactId("mngeclipse-2110");
-    archetype.setVersion("1.0");
-    archetype.setRepository(new File("repositories/testrepo").getAbsoluteFile().toURI().toASCIIString());
-    Collection<IProject> projects = createProjectsFromArchetype("mngeclipse-2110", new MavenArchetype(archetype), null);
-    assertEquals(1, projects.size());
-    IProject project = projects.iterator().next();
-    assertTrue(project.isAccessible());
+    File templateSettings = new File("settings_archetype_repo.xml").getCanonicalFile().getAbsoluteFile();
+    String interpolated = Files.readString(templateSettings.toPath()).replace("${repoUrl}",
```
[the remark this answers is not in the document]
Comment (laeubi) [reply org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java:183]: I have no idea why relative do not work in this case, and its better than delete the test I think ;-)

**Comment by laeubi** [comment]
> It was necessary because of [ARCHETYPE-519](https://issues.apache.org/jira/browse/ARCHETYPE-519), wasn't it?

Yes it is documented here: https://maven.apache.org/archetype/maven-archetype-plugin/archetype-repository.html

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java (hunk 1 above)
Comment (laeubi) [reply org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java:185]: Yes, but the rule will apply to all methods, I'm not sure if that worth it...

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (HannesWell) [reply org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java:183]: > I have no idea why relative do not work in this case, and its better than delete the test I think ;-)

Hmm. To bad, but as long as it works and it is not too complicated, yes it is definitely better than deleting. 👍🏽

**Pull request merged**